### PR TITLE
Fixes #20924 - check credentials before doing any changes

### PR DESCRIPTION
**Pull request by evgeni**
depends on https://github.com/Katello/katello-packaging/pull/526

### CODE REVIEW

File: katello/katello-change-hostname [hunk 1]
```diff
@@ -170,6 +171,12 @@ def precheck
     fail_with_message("There is a problem with the server, please check 'hammer ping'")
   end
 
+  STDOUT.puts "\nChecking credentials"
```
Comment (johnpmitsch) [inline katello/katello-change-hostname:174]: I was trying to think if there is a more generic command we can use, but can't really find one! Maybe @akofink can think of one (we are just trying to check if the username/password credentials are correct)

File: katello/katello-change-hostname [hunk 2]
```diff
@@ -170,6 +171,12 @@ def precheck
...
+  hammer_cmd("capsule list")
+  unless $?.exitstatus == 0
```
Comment (johnpmitsch) [inline katello/katello-change-hostname]: can we use `$?.success?` syntax here? I use it in other places in the script.

**Review by johnpmitsch — CHANGES_REQUESTED** [review]
just a couple small comments

File: katello/katello-change-hostname (hunk 1 above)
Comment (akofink) [reply katello/katello-change-hostname:174]: `hammer ping` ?

Comment (evgeni) [reply katello/katello-change-hostname:174]: `hammer ping` does not require credentials, and it does not error out for wrong ones either.
we could do `hammer user info --name #{@options[:username]}`, but that feels equally ugly.

File: katello/katello-change-hostname (hunk 2 above)
Comment (evgeni) [reply katello/katello-change-hostname]: Yes we can!

File: katello/katello-change-hostname (hunk 1 above)
Comment (johnpmitsch) [reply katello/katello-change-hostname:174]: I'm fine with `capsule list`, since we will run that later.

**Review by johnpmitsch — APPROVED** [review]
Works well
```
[root@changed34 ~]# ./katello-change-hostname -u admin -p notmypassword changedagain.example.com

Checking hostname validity

Checking overall health of server

Checking credentials
Invalid username or password
There is a problem with the credentials, please retry
```
and with the actual password
```
[root@changed34 ~]# ./katello-change-hostname -u admin -p changeme$ -yd changedagain.example.com
```
worked as expected :+1:

**Pull request merged**